rdmo/projects/imports.py

```diff
@@ -100,11 +101,21 @@ def check(self):
     def process(self):
         if self.current_project is None:
             catalog_uri = get_uri(self.root.find('catalog'), self.ns_map)
+
+            available_catalogs = Catalog.objects.filter_current_site() \
+                                                .filter_group(self.request.user) \
+                                                .filter_availability(self.request.user) \
+                                                .order_by('order')
+
             try:
-                self.catalog = Catalog.objects.all().get(uri=catalog_uri)
+                self.catalog = available_catalogs.get(uri=catalog_uri)
             except Catalog.DoesNotExist:
                 log.info('Catalog not in db. Created with uri %s', catalog_uri)
-                self.catalog = Catalog.objects.all().first()
+                self.catalog = available_catalogs.first()
+
+            if self.catalog is None:
+                log.info('No catalog is available.')
+                raise ValidationError('No catalog is available.')
```

Comment:
the front end does not handle this error yet..

Reply:
Yes, but this should not happen anyway. No worries.